extensions/EntityExtension/src/main/kotlin/com/typewritermc/entity/entries/activity/RandomPatrolActivity.kt

```diff
@@ -0,0 +1,133 @@
+
+    override val currentProperties: List<EntityProperty>
+        get() = childActivity.currentProperties
+}
+
+private class RandomPatrolActivity(
+    private val roadNetwork: Ref<RoadNetworkEntry>,
+    private val nodes: List<RoadNodeId>,
+    private val startLocation: PositionProperty,
+) : EntityActivity<ActivityContext>, KoinComponent {
+    private var currentLocationIndex = 0
+    private var activity: EntityActivity<in ActivityContext>? = null
+
+    fun refreshActivity(context: ActivityContext, network: RoadNetwork) {
+        val randomNodeId = nodes.randomOrNull()
+            ?: throw IllegalStateException("Could not find any node in the nodes list for the random patrol activity.")
+        val targetNode = network.nodes.find { it.id == randomNodeId }
+            ?: throw IllegalStateException("Could not find any node in the nodes list for the random patrol activity.")
+
+        activity?.dispose(context)
+        activity = NavigationActivity(PointToPointGPS(
+            roadNetwork,
+            { currentPosition.toBukkitLocation() }) {
+            targetNode.location
+        }, currentPosition
+        )
+        activity?.initialize(context)
+    }
+
+    override fun initialize(context: ActivityContext) = setup(context)
+
+    private fun setup(context: ActivityContext) {
+        val network =
+            KoinJavaComponent.get<RoadNetworkManager>(RoadNetworkManager::class.java).getNetworkOrNull(roadNetwork)
+                ?: return
+
+        // Get the closest node to the start location
+        val closestNode = network.nodes
+            .filter { it.id in nodes }
+            .minByOrNull { it.location.distanceSqrt(startLocation.toBukkitLocation()) ?: Double.MAX_VALUE }
+            ?: throw IllegalStateException("Could not find any node in the nodes list for the random patrol activity.")
+
+        val index = nodes.indexOf(closestNode.id)
+        currentLocationIndex = (index + 1) % nodes.size
+        refreshActivity(context, network)
+    }
+
+    override fun tick(context: ActivityContext): TickResult {
+        if (activity == null) {
+            setup(context)
+            return TickResult.CONSUMED
+        }
+
+        val network =
+            KoinJavaComponent.get<RoadNetworkManager>(RoadNetworkManager::class.java).getNetworkOrNull(roadNetwork)
+                ?: return TickResult.IGNORED
+
+        val result = activity?.tick(context)
+        if (result == TickResult.IGNORED) {
+            refreshActivity(context, network) // Select a random node
+        }
+
+        return TickResult.CONSUMED
+    }
+
+    override fun dispose(context: ActivityContext) {
+        activity?.dispose(context)
+        activity = null
+    }
+
+    override val currentPosition: PositionProperty
+        get() = activity?.currentPosition ?: startLocation
+
+    override val currentProperties: List<EntityProperty>
+        get() = activity?.currentProperties?.filter { it !is PositionProperty } ?: listOf()
```

Comment:
@Kerzinator24 I think the AI is correct here. I did the filtering in the `LookYawPith` because I added the current location after it again. 

```kotlin LookYawPitchAcitivity
    override val currentProperties: List<EntityProperty>
        get() = childActivity.currentProperties.filter { it !is PositionProperty } + currentPosition
```

You see I added the currentPosition again because I change it there.

Since you just pass the location onwards, you don't need the filter here.